src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -563,11 +564,29 @@ public static MsBuildToolset GetMsBuildToolset(string userVersion, IConsole cons
         /// <param name="getMsBuildPathInPathVar">delegate to provide msbuild exe discovered in path environemtnb var/s
         /// (using a delegate allows for testability)</param>
         /// <returns>directory to use for msbuild exe</returns>
+        [Obsolete("Use GetMsBuildDirectoryInternal(string, IEnumerable<MsBuildToolset>, Func<IEnvironmentVariableReader, string>) instead")]
```

Comment:
nuget.exe is ILMerged and not shipped as a library. 

You can remove this method.

Reply:
Removed.